### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by cpjolicoeur**
* Properly capitalize TypeScript (and link the first occurrence to the website. 
* Renamed the variables in the example code to be a little less "coder generic"
* Added proper Markdown spacing around code blocks
* Added code block syntax highlighting markers

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,33 +31,35 @@ interface Pet {
 ```
 
 And we've got some data about current client's pets which is stored as JSON:
-```
-const pet1: Pet = JSON.parse('{"name":"Lyle","species":"Crocodile","isCute":true}')
-const pet2: Pet = JSON.parse('{"name":"Bullwinkle","age":59}')
+
+```javascript
```
Comment (mulias) [inline README.md]: this can be a typescript block, since it has type annotations.

File: README.md [hunk 2]
```diff
@@ -30,33 +31,35 @@ interface Pet {
...
-Unfortunately typescript doesn't provide a good built-in way to deal with this
-issue. Providing run-time type information is one of typescript's
+Unfortunately TypeScript doesn't provide a good built-in way to deal with this
+issue. Providing run-time type information is one of TypeScript's
 [non-goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals#non-goals),
-and our web app is too important to risk using a forked version of typescript
+and our web app is too important to risk using a forked version of TypeScript
 with that added functionality.
-[Type guards](https://basarat.gitbooks.io/typescript/docs/types/typeGuard.html)
+[Type guards](https://basarat.gitbooks.io/TypeScript/docs/types/typeGuard.html)
```
Comment (mulias) [inline README.md]: not strictly correct, but the link still works so ¯\\\_(ツ)\_/¯

Comment (cpjolicoeur) [reply README.md]: oh, my bad.  that was a global find/replace.  I can change it

**Comment by mulias** [comment]
@cpjolicoeur want me to update this quick?

**Comment by cpjolicoeur** [comment]
@mulias sorry for the delay, yeah if you want to.  Not sure when I'll get back on this to update it this week, so if you want to just take it over, be my guest.

sorry about that

**Pull request merged**